test/approx1.cpp

```diff
@@ -840,3 +840,83 @@ TEST(Approx1, CPPEmptyPosAndInput)
     ASSERT_TRUE(pos.isempty());
     ASSERT_TRUE(interp.isempty());
 }
+
+TEST(Approx1, UseNullInitialOutput) {
+    float h_in[3] = {10, 20, 30};
+    dim_t h_in_dims = 3;
+
+    af_array in = 0;
+    ASSERT_SUCCESS(af_create_array(&in, &h_in[0], 1, &h_in_dims, f32));
+
+    float h_pos[5] = {0.0, 0.5, 1.0, 1.5, 2.0};
+    dim_t h_pos_dims = 5;
+    af_array pos = 0;
+    ASSERT_SUCCESS(af_create_array(&pos, &h_pos[0], 1, &h_pos_dims, f32));
+
+    af_array out = 0;
+    ASSERT_SUCCESS(af_approx1(&out, in, pos, AF_INTERP_LINEAR, 0));
+
+    ASSERT_FALSE(out == 0);
+}
+
+TEST(Approx1, UseExistingOutputArray) {
+    float h_in[3] = {10, 20, 30};
+    dim_t h_in_dims = 3;
+
+    af_array in = 0;
+    ASSERT_SUCCESS(af_create_array(&in, &h_in[0], 1, &h_in_dims, f32));
+
+    float h_pos[5] = {0.0, 0.5, 1.0, 1.5, 2.0};
+    dim_t h_pos_dims = 5;
+    af_array pos = 0;
+    ASSERT_SUCCESS(af_create_array(&pos, &h_pos[0], 1, &h_pos_dims, f32));
+
+    dim_t h_out_dims = 5;
+    af_array out_ptr = 0;
+    ASSERT_SUCCESS(af_create_handle(&out_ptr, 1, &h_out_dims, f32));
+    af_array out_ptr_copy = out_ptr;
+    ASSERT_SUCCESS(af_approx1(&out_ptr, in, pos, AF_INTERP_LINEAR, 0));
+
+    ASSERT_EQ(out_ptr_copy, out_ptr);
+    // Verify that the original ASSERT_SUCCESS(af_array does contain the results
+    ASSERT_ARRAYS_EQ(out_ptr_copy, out_ptr);
+}
+
+TEST(Approx1, UseExistingOutputSlice) {
+    float h_in[3] = {10, 20, 30};
+    dim_t h_in_dims = 3;
+
+    af_array in = 0;
+    ASSERT_SUCCESS(af_create_array(&in, &h_in[0], 1, &h_in_dims, f32));
+
+    float h_pos[5] = {0.0, 0.5, 1.0, 1.5, 2.0};
+    dim_t h_pos_dims = 5;
+    af_array pos = 0;
+    ASSERT_SUCCESS(af_create_array(&pos, &h_pos[0], 1, &h_pos_dims, f32));
+
+    float h_out[15] = {1.0, 1.5, 2.0, 2.5, 3.0,
+                       4.0, 4.5, 5.0, 5.5, 6.0,
+                       7.0, 7.5, 8.0, 8.5, 9.0};
+    dim_t h_out_dims[2] = {5, 3};
+    af_array out = 0;
+    ASSERT_SUCCESS(af_create_array(&out, &h_out[0], 2, &h_out_dims[0], f32));
+    af_seq idx_dim1 = {1, 1, 1}; // get slice 1 of dim1
+    af_seq idx[2] = {af_span, idx_dim1};
+    af_array out_slice = 0;
+    ASSERT_SUCCESS(af_index(&out_slice, out, 2, &idx[0]));
+    ASSERT_SUCCESS(af_approx1(&out_slice, in, pos, AF_INTERP_LINEAR, 0));
+
+    dim_t nelems = 0;
+    ASSERT_SUCCESS(af_get_elements(&nelems, out));
+    vector<float> h_out_approx(nelems);
+    ASSERT_SUCCESS(af_get_data_ptr(&h_out_approx.front(), out));
+
+    float h_gold_arr[5] = {10.0, 15.0, 20.0, 25.0, 30.0};
+
+    // Check slice 1 of dim2 (elements 5 to 9 of h_out_approx) to see if they
+    // contain ASSERT_SUCCESS(af_approx1's results (without additional intermediate processing
+    // like indexing)
+    for (int i = 0; i < 5; ++i) {
```

Comment:
You should check all values because we don't want other values to change. you could just copy the original array and update the values that changed.